### ISSUE TITLE
OOM问题，参考lerobot_example，8卡A100*40GB

### ISSUE THREAD

**Issue by SoonSYJ** [X-Square-Robot/wall-x#15]
作者您好，可否分享一下您的机器配置。我这边batch_size_per_gpu=1当跑到gradient_accumulation_step就会OOM。不知该如何优化？

<img width="951" height="666" alt="Image" src="https://github.com/user-attachments/assets/e15fd79a-0c34-4cdc-ac92-6104f5089b3a" />

**Comment by StarrickLiu** [comment]
Thank you for reporting this OOM issue. It's quite common to encounter OOM with 8x A100 40GB GPUs even at batch_size_per_gpu=1, primarily because the current implementation doesn't utilize more efficient parallelization strategies.

**Optimization approaches you can try:**

1. **Gradient Checkpointing**: Use `torch.utils.checkpoint` for activation recomputation to trade compute time for memory savings
   
2. **Distributed Training Optimization**:
   - Enable DeepSpeed ZeRO Stage 2 in your accelerate configuration
   - Or use PyTorch FSDP (Fully Sharded Data Parallel)

**Important Notes:**
- These modifications may require code adjustments, so please verify training accuracy after implementation
- Official support for FSDP + ZeRO2 is coming soon, which will significantly reduce multi-GPU memory overhead
